/*
 *  Copyright (C) 2020-2022 Embedded AMS B.V. - All Rights Reserved
 *
 *  This file is part of Embedded Proto.
 *
 *  Embedded Proto is open source software: you can redistribute it and/or 
 *  modify it under the terms of the GNU General Public License as published 
 *  by the Free Software Foundation, version 3 of the license.
 *
 *  Embedded Proto  is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with Embedded Proto. If not, see <https://www.gnu.org/licenses/>.
 *
 *  For commercial and closed source application please visit:
 *  <https://EmbeddedProto.com/license/>.
 *
 *  Embedded AMS B.V.
 *  Info:
 *    info at EmbeddedProto dot com
 *
 *  Postal address:
 *    Johan Huizingalaan 763a
 *    1066 VH, Amsterdam
 *    the Netherlands
 */

// This file is used to test the wire formatter class.

syntax = "proto3";

enum Test_Enum {
  ZERO  = 0;
  ONE   = 1;
  TWO   = 2;
  ONEHUNDERTTWENTYSEVEN = 127;
  ONEHUNDERTTWENTYEIGHT = 128;
  TWOHUNDERDFIFTYFIVE   = 255;
  TWOBILLION = 2000000000;
}

message Test_Simple_Types {
  int32       a_int32     = 1;
  int64       a_int64     = 2;
  uint32      a_uint32    = 3;
  uint64      a_uint64    = 4; 
  sint32      a_sint32    = 5;
  sint64      a_sint64    = 6; 
  bool        a_bool      = 7;
  Test_Enum   a_enum      = 8;
  fixed64     a_fixed64   = 9;
  sfixed64    a_sfixed64  = 10;
  double      a_double    = 11;
  fixed32     a_fixed32   = 12;
  sfixed32    a_sfixed32  = 13;
  float       a_float     = 14;

  enum Nested_Enum {
    NE_A = 0;
    NE_B = 1;
    NE_C = 2;
  }

  Nested_Enum a_nested_enum = 15;
}
